drivers/serial/uart_nrfx_uarte.c

```diff
@@ -1044,11 +1044,11 @@ static void rxstarted_isr(const struct device *dev)
 		.type = UART_RX_BUF_REQUEST,
 	};
 	user_callback(dev, &evt);
-	if (data->async->rx_timeout != SYS_FOREVER_MS) {
+	if (data->async->rx_timeout != SYS_FOREVER_US) {
 		data->async->rx_timeout_left = data->async->rx_timeout;
 		k_timer_start(&data->async->rx_timeout_timer,
-			      K_MSEC(data->async->rx_timeout_slab),
-			      K_MSEC(data->async->rx_timeout_slab));
+			      K_USEC(data->async->rx_timeout_slab),
+			      K_USEC(data->async->rx_timeout_slab));
```

Comment:
Nit: two comments, in lines [929](https://github.com/zephyrproject-rtos/zephyr/blob/9208b14fde0e71b9c5bd9237c7e9af88f18a6022/drivers/serial/uart_nrfx_uarte.c#L929) and [999](https://github.com/zephyrproject-rtos/zephyr/blob/9208b14fde0e71b9c5bd9237c7e9af88f18a6022/drivers/serial/uart_nrfx_uarte.c#L999), still mention "ms" as the RX timeout unit.

Also the calculation of `rx_timeout_slab`:
https://github.com/zephyrproject-rtos/zephyr/blob/9208b14fde0e71b9c5bd9237c7e9af88f18a6022/drivers/serial/uart_nrfx_uarte.c#L809-L811
 should be updated, as the minimum value there is set for milliseconds.